modules/business-text/business-text.service.js

```diff
@@ -37,32 +39,122 @@ class BusinessTextService {
+      }
+    } else {
+      throw new Error(BUSINESS_TEXT_NOT_FOUND);
+    }
+
+    const text = await BusinessText.findByIdAndUpdate(id, newPage, {
       new: true,
     });
 
     return text || null;
   }
 
-  async addBusinessText(data) {
-    const pages = await this.checkBusinessTextExistByCode(data);
+  async addBusinessText(businessText, files) {
+    const existingPages = await this.checkBusinessTextExistByCode(businessText);
 
-    if (pages.length) {
+    if (existingPages.length) {
       throw new Error(BUSINESS_TEXT_ALREADY_EXIST);
     }
-    return new BusinessText(data).save();
+
+    let newPage = '';
+    if (files.length) {
+      newPage = await this.pageImageReplaceHandler(businessText, files);
+    }
+
+    return new BusinessText(newPage || businessText).save();
   }
 
   async deleteBusinessText(id) {
+    const oldPage = await this.getBusinessTextById(id);
+
+    if (oldPage) {
+      const imagesToDelete = this.imageFinder(oldPage);
+
+      if (imagesToDelete.length) {
+        await this.deleteNoNeededImages(imagesToDelete);
+      }
+    } else {
+      throw new Error(BUSINESS_TEXT_NOT_FOUND);
+    }
+
     const businessText = await BusinessText.findByIdAndDelete(id);
     if (businessText) {
       return businessText;
     }
-    throw new Error(BUSINESS_TEXT_NOT_FOUND);
   }
 
   async checkBusinessTextExistByCode(data) {
     return await BusinessText.find({ code: data.code });
   }
+
+  async pageImageReplaceHandler(page, files) {
+    const fileNames = files.map(({ file }) => file.filename);
+
+    const uploadResult = await uploadFiles(files);
+    const imagesResults = await Promise.allSettled(uploadResult);
+
+    const updatedPage = page;
+    let newUkText = '';
+    let newEnText = '';
+
+    fileNames.forEach((name, i) => {
+      const regExp = new RegExp(`src=""(?=.*alt="${name}")`, 'g');
+      const replacer = `src="${imagesResults[i].value.prefixUrl}${imagesResults[i].value.fileNames.small}"`;
+
+      newUkText = newUkText
+        ? newUkText.replace(regExp, replacer)
+        : updatedPage.text[0].value.replace(regExp, replacer);
+
+      newEnText = newEnText
+        ? newEnText.replace(regExp, replacer)
+        : updatedPage.text[1].value.replace(regExp, replacer);
+    });
+
+    updatedPage.text[0].value = newUkText || page.text[0].value;
+    updatedPage.text[1].value = newEnText || page.text[1].value;
+
+    return updatedPage;
+  }
+
+  async deleteNoNeededImages(images) {
+    const regExp = new RegExp(
+      `(?<=src="${process.env.IMAGE_LINK}[a-z]+_).*?(?=")`
+    );
+    const valuesToDelete = images
+      .flatMap(img => img.match(regExp))
+      .flatMap(id => [
+        `large_${id}`,
+        `medium_${id}`,
+        `small_${id}`,
+        `thumbnail_${id}`,
+      ]);
+
+    await Promise.allSettled(await deleteFiles(valuesToDelete));
+  }
+
+  imageFinder(page) {
+    const images = page.text
+      .flatMap(({ value }) => value.match(/<img([\w\W]+?)>/g))
```

Comment:
It is better to create a separate variable for matched values because code with `flatMap` is harder to understand